src/Semaphore.sol

```diff
@@ -164,4 +187,29 @@ contract Semaphore is IWorldID, SemaphoreCore, Verifier, SemaphoreGroups {
 
         manager = newManager;
     }
+    ///////////////////////////////////////////////////////////////////////////////
+    ///                          VIEW FUNCTIONS                                ///
+    //////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks if a given root value is valid and has been added to the root history
+    /// @param root  The root of a given identity group
+    /// @param groupId The id of the group
+    function checkValidRoot(uint256 root, uint256 groupId) public view returns (bool) {
```

Comment:
Same as above, this can probably be done with a single state read for latestRoots

Reply:
I think that when we check validity here we want to know more information about this specific root given. Since the latest root is evergreen, this check alone is sufficient to know if it's up to date (valid), but not in telling us anything about a possible previous state of a given root.

We might not care about all that information though, just my 2cents on why we would want to do additional comparisons.

Reply:
I see that we can accommodate both concerns, updated accordingly. I think I interpreted your comment to suggesting it being the only check.